### PR TITLE
travis: add more architectures + enable Coverity

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian@brauner.io>

### CODE REVIEW

**Comment by jubalh** [comment]
AFAIK they only allow a certain number of builds (per day?) and for that reason they advise to have a dedicated branch and not run on master, if I remember correctly.

**Comment by brauner** [comment]
> AFAIK they only allow a certain number of builds (per day?) and for that reason they advise to have a dedicated branch and not run on master, if I remember correctly.

True, but we don't have that many merges that we need to worry about that?

**Comment by brauner** [comment]
> > AFAIK they only allow a certain number of builds (per day?) and for that reason they advise to have a dedicated branch and not run on master, if I remember correctly.
> 
> True, but we don't have that many merges that we need to worry about that?

Sincere question. :)

**Comment by jubalh** [comment]
> True, but we don't have that many merges that we need to worry about that?

Probably :) I don't know the exact number of requests. But I thought if for some time you don't merge anything. And then have some free time slot and merge 20 requests on one day, maybe it already is too much.

So I don't know just wanted to mention it in case you were not aware of the limitation :)

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,38 @@
...
-  apt:
-    packages:
-      - autopoint
-      - xsltproc
+  coverity_scan:
+    project:
+      name: "shadow-maint/shadow"
+      description: "Upstream shadow utils tree"
+    notification_email: christian.brauner@ubuntu.com,serge@hallyn.com
+    build_command_prepend: "./autogen.sh --without-selinux --disable-man"
```
Comment (hallyn) [inline .travis.yml:33]: Unrelated to this change, but I've beenw ondering whether we should not do --disable-man.

**Comment by hallyn** [comment]
Thanks.

File: .travis.yml (hunk 1 above)
Comment (brauner) [reply .travis.yml:33]: We probably should...

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,38 @@
...
+
+env:
+  global:
+   # The next declaration is the encrypted COVERITY_SCAN_TOKEN, created
+   #   via the "travis encrypt" command using the project repo's public key
+   - secure: "D3C+O5UjN8Oe0FROMk5yPRU0he4UNAaHdxwSVG2373ImzYIs3pIvre1kXzeQSajbXCRKiKE8p2U0TooIRZAo/JeaZDgb4dBhij4c4/Qv8MpFCTC1W7cgo6UeOQeZwdfqp7v9Zz5ubh6RW68bmSM2naqQim1zuD1vDSGy4Wir7z4="
+
+before_install:
+  - sudo apt-get update -qq
+  - sudo apt-get -y install -qq automake autopoint xsltproc libselinux1-dev gettext
```
Comment (hallyn) [inline .travis.yml:24]: Oh, ...  switching from the yaml based list of packages to install seems like maybe a regression?  Arguable, I guess.

**Pull request merged**